tests/gumjs/script-mac.m

```diff
@@ -147,6 +149,123 @@
   }
 }
 
+@interface FridaTest1 : NSObject
++ (int)foo_;
++ (int)fooBar_;
++ (int)fooBar:(int)a;
++ (int):(int)a;
++ (int):(int)a :(int)b;
+@end
+
+@implementation FridaTest1
++ (int)foo_ {
+  return 1;
+}
++ (int)fooBar_ {
+  return 2;
+}
++ (int)fooBar:(int)a {
+  return 3;
+}
++ (int):(int)a {
+  return 4;
+}
++ (int):(int)a :(int)b {
+  return 5;
+}
+@end
+
+SCRIPT_TESTCASE (methods_with_weird_names_can_be_invoked)
+{
+  @autoreleasepool
+  {
+    COMPILE_AND_LOAD_SCRIPT (
+        "var FridaTest1 = ObjC.classes.FridaTest1;"
+        "var methodNames = ['foo_', 'fooBar_', 'fooBar:', ':', '::'];"
+        "var args = [0, 0, 1, 1, 2];"
+        "for (var i = 0; i < methodNames.length; i++) {"
+            "var m = FridaTest1['+ ' + methodNames[i]];"
+            "var val = m.apply(FridaTest1, args[i] == 0? []: args[i] == 1? [0]: [0, 0]);"
+            "send(val == i + 1);"
+        "}");
+
+    for (int i = 0; i < 5; i++) {
```

Comment:
Style nitpick for here and below: `i !== 5` and mis-aligned curly braces. We tend to omit the block for single statements though, as long as it's not a massively long statement spanning multiple lines.